app/services/person_constructor.rb

```diff
@@ -0,0 +1,82 @@
+class PersonConstructor
+  KEY_NAME_MAPPINGS = {
+    address: :address_1,
+    zip: :zip_code,
+  }
+  OLD_REMOTE_PARAMS = [
+    remote_fields: PersonWithRemoteFields::REMOTE_PARAMS
+  ]
+
+  def self.permitted_params
```

Comment:
Yeah, I thought you might say that. I have been trying to favor methods over variables since reading POODR. It allows for more flexibility as things change in the future. And I think it provides a clearer interface. I was going to have all of these new classes use methods instead of variables for any data that might be shared between classes, but it started to seem like overkill. So I guess the end result looks a little inconsistent... My excuse was that PersonConstructor is very closely bound to the implementation of the classes that it's updating, while any controller, etc., should be somewhat removed from knowing anything about the internals of PersonConstructor and its associated classes.

I can change it to a constant if you think it's a problem, though